src/Xrefcheck/Core.hs

```diff
@@ -71,16 +71,18 @@ instance Given ColorMode => Buildable Position where
 
 -- | Full info about a reference.
 data Reference = Reference
-  { rName   :: Text
+  { rName :: Text
     -- ^ Text displayed as reference.
-  , rLink   :: Text
+  , rLink :: Text
     -- ^ File or site reference points to.
   , rAnchor :: Maybe Text
     -- ^ Section or custom anchor tag.
-  , rPos    :: Position
+  , rPos :: Position
     -- ^ Position in source file.
-  , rInfo   :: ReferenceInfo
+  , rInfo :: ReferenceInfo
     -- ^ More info about the link.
+  , rIsSymlink :: Bool
+    -- ^ Whether ir corresponds to a symlink.
```

Comment:
This can be better modelled, but it is temporary until we discuss the symlinks output in this draft PR.